src/Components/Components/test/Routing/RouterTest.cs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Routing;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Microsoft.Extensions.DependencyModel;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components.Test.Routing
+{
+    public class RouterTest
+    {
+        [Fact]
+        public void CanRunOnNavigateAsync()
+        {
+            // Arrange
+            var router = CreateMockRouter();
+            var called = false;
+            async Task OnNavigateAsync(NavigationContext args)
+            {
+                await Task.CompletedTask;
+                called = true;
+            }
+            router.Object.OnNavigateAsync = new EventCallbackFactory().Create<NavigationContext>(router, OnNavigateAsync);
+
+            // Act
+            router.Object.RunOnNavigateWithRefreshAsync("http://example.com/jan", false);
```

Comment:
Why is it ok to not await these tasks?

Reply:
Starting the task without awaiting better simulates how the invocation ends up being made when we call it from `OnLocationChanged` in the router. This presents an issue when the tests are running in CI though since we don't wait for the async tasks to complete before validating what they did...

Reply:
Generally, when Tasks aren't directly awaited in a test, I still like storing the Task and asserting that it's incomplete. Later on in the test when I expect the Task to be completed (if ever), I then assert the Task is complete and verify it has the expected result (if any).

In this case, RunOnNavigateWithRefreshAsync should be returning a completed Task, right? If that's the case, I would just "await" it.

Reply:
Yes, `RunOnNavigateWithRefreshAsync` returns a completed task. Updated this to use `await`.